File: webapp-net/DynamicDocumentation/Mapping/ModelBuilder.cs
using System;
using Sdl.Web.Common.Models;
using Sdl.Web.DataModel;
using Sdl.Web.Tridion.Mapping;

namespace Sdl.Web.Modules.DynamicDocumentation.Mapping
{
    /// <summary>
    /// Model Builder
    /// 
    /// Remaps 'Ish' mvc area to DynamicDocumentation
    /// 
    /// Add to Web.Config:
    ///  <modelBuilderPipeline>
    ///     <add type = "Sdl.Web.Tridion.Mapping.DefaultModelBuilder, Sdl.Web.Tridion" />
    ///     <add type = "Sdl.Web.Modules.DDWebAppReact.Mapping.DDWebAppModelBuilder, Sdl.Web.Modules.DDWebAppReact" />
    ///     ...
    ///  </modelBuilderPipeline>
    /// </summary>
    public class ModelBuilder : IPageModelBuilder, IEntityModelBuilder
    {
        public void BuildPageModel(ref PageModel pageModel, PageModelData pageModelData, bool includePageRegions,
            Common.Configuration.Localization localization)
        {
            DataModel.MvcData mvcData = pageModelData.MvcData;
            RemapMvcAreaName(ref mvcData);
        }

        public void BuildEntityModel(ref EntityModel entityModel, EntityModelData entityModelData, Type baseModelType,
            Common.Configuration.Localization localization)
        {
            DataModel.MvcData mvcData = entityModelData.MvcData;
            RemapMvcAreaName(ref mvcData);
        }

        private void RemapMvcAreaName(ref DataModel.MvcData mvcData)
        {
            if (mvcData != null && mvcData.AreaName != null &&
               mvcData.AreaName.Equals("Ish"))
            {
                mvcData.AreaName = DynamicDocumentationModuleAreaRegistration.AREA_NAME;
            }
        }
    }
}

File: webapp-java/dxa-module-dynamic-documentation/src/main/java/com/sdl/dxa/modules/dd/mapping/ModelBuilder.java
package com.sdl.dxa.modules.dd.mapping;

import com.sdl.dxa.api.datamodel.model.EntityModelData;
import com.sdl.dxa.api.datamodel.model.MvcModelData;
import com.sdl.dxa.api.datamodel.model.PageModelData;
import com.sdl.dxa.modules.dd.DDModuleInitializer;
import com.sdl.dxa.tridion.mapping.EntityModelBuilder;
import com.sdl.dxa.tridion.mapping.PageModelBuilder;
import com.sdl.webapp.common.api.mapping.semantic.SemanticMappingException;
import com.sdl.webapp.common.api.model.EntityModel;
import com.sdl.webapp.common.api.model.PageModel;
import com.sdl.webapp.common.exceptions.DxaException;
import org.jetbrains.annotations.Nullable;
import org.springframework.stereotype.Service;

/**
  * Model Builder
  *
  * Remaps 'Ish' mvc area to DynamicDocumentation
 */
@Service
public class ModelBuilder implements EntityModelBuilder, PageModelBuilder {

    private void RemapMvcAreaName(MvcModelData mvcData) {
        if (mvcData != null && mvcData.getAreaName() != null &&
                mvcData.getAreaName().equals("Ish")) {
            mvcData.setAreaName(DDModuleInitializer.DYNAMIC_DOCUMENTATION);
        }
    }

    @Override
    public <T extends EntityModel> T buildEntityModel(@Nullable T t, EntityModelData entityModelData, @Nullable Class<T> aClass) throws DxaException {
        MvcModelData mvcData = entityModelData.getMvcData();
        RemapMvcAreaName(mvcData);
        return t;
    }

    @Override
    public @Nullable PageModel buildPageModel(@Nullable PageModel pageModel, PageModelData pageModelData) throws SemanticMappingException {
        MvcModelData mvcData = pageModelData.getMvcData();
        RemapMvcAreaName(mvcData);
        return pageModel;
    }

    @Override
    public int getOrder() {
        return 0;
    }
}
